lib/handlers/error-pages.js

```diff
@@ -9,12 +9,36 @@ function handler (err, req, res, next) {
 
   var ldp = req.app.locals.ldp
 
-  if (err.status === 401 &&
-      req.accepts('text/html') &&
-      ldp.auth === 'oidc') {
-    debug('401 error - redirect to Select Provider')
-    res.status(err.status)
-    redirectToLogin(req, res, next)
+  if (ldp.auth === 'oidc') {
+    let statusCode = err.status || err.statusCode || 500
+
+    res.status(statusCode)
+
+    let errorParams = {
+      realm: ldp.host.serverUri,
+      scope: 'openid',
+      error: err.error,
+      error_description: err.error_description,
+      error_uri: err.error_uri
+    }
+
+    let challengeParams = Object.keys(errorParams)
+      .filter(key => !!errorParams[key])
+      .map(key => `${key}="${errorParams[key]}"`)
+      .join(', ')
+
+    res.set('WWW-Authenticate', 'Bearer ' + challengeParams)
```

Comment:
This header is meant to be used with `401`s, but it's being used here regardless of the error code.